core/lib/types/PathSeq.agda

```diff
@@ -296,60 +296,68 @@ module _ {i} {A : Type i} {a a' : A} where
   _∙ₛ_ : {s t u : a =-= a'} → s =ₛ t → t =ₛ u → s =ₛ u
   _∙ₛ_ (=ₛ-intro p) (=ₛ-intro q) = =ₛ-intro (p ∙ q)
 
+  {-
+    Note: While this enables more succinct chains of equations in comparison to
+    chains using _=↯=⟨_&_&_&_⟩_ (since it avoids having to spell out the target
+    subsequence), it is also results in significantly (~ one order of magnitude)
+    slower type checking. Therefore, this function should only be used for
+    developing new proofs, not to simplify old proofs.
+  -}
```

Comment:
Do you think it's due to the instance argument search or just the unification (for implicit arguments)?

Reply:
I measured the time it takes to check `core/lib/two-semi-categories/FunCategory.agda` (including loading the dependencies)

* with `=ₛ` with instance arguments: 25.2s (1st run), 21.2s (2nd run), 21.5s (3rd run)
* with `=ₛ` with explicit arguments: 13.9s, 13.2s, 13.4s
* with `=↯=`: 5.8s, 6.0s, 5.9s

So it seems like unification and instance resolution are each responsible for about half of the increase in checking time.

Reply:
If I only check `CupProduct.agda`...
```
$ agda --version
Agda version 2.5.4-120f3dd-dirty
$ rm theorems/cohomology/CupProduct.agdai
$ /bin/time -v agda theorems/cohomology/CupProduct.agda
Checking cohomology.CupProduct (/home/favonia/documents/records/code/utt-agda/theorems/cohomology/CupProduct.agda).
	Command being timed: "agda theorems/cohomology/CupProduct.agda"
	User time (seconds): 169.71
	System time (seconds): 2.27
	Percent of CPU this job got: 99%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 2:52.96
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 2500484
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 617781
	Voluntary context switches: 1
	Involuntary context switches: 1285
	Swaps: 0
	File system inputs: 0
	File system outputs: 1008
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
```